src/machines/__tests__/auth-token-polling-machine.test.ts

```diff
@@ -0,0 +1,122 @@
+import {authTokenPollingMachine} from '../auth-token-polling-machine'
+import {interpret} from 'xstate'
+import {SimulatedClock} from 'utils/test/simulated-clock'
+
+function sleep(time: number) {
+  return new Promise((resolve) => {
+    return setTimeout(resolve, time)
+  })
+}
+
+test('it starts polling immediately', () => {
+  const pollingService = interpret(
+    authTokenPollingMachine.withConfig({
+      services: {
+        requestAuthToken: () => Promise.resolve(),
+      },
+    }),
+  )
+
+  pollingService.start()
+
+  expect(pollingService.state).toMatchState({pending: 'polling'})
+})
+
+test('it invokes the requestAuthToken service', () => {
+  const mockedFunc = jest.fn()
+
+  const pollingService = interpret(
+    authTokenPollingMachine.withConfig({
+      services: {
+        requestAuthToken: () => {
+          mockedFunc()
+
+          return Promise.resolve()
+        },
+      },
+    }),
+  )
+
+  pollingService.start()
+
+  expect(mockedFunc).toHaveBeenCalled()
+})
+
+test('it assigns the authToken to context when received', async () => {
+  const pollingService = interpret(
+    authTokenPollingMachine.withConfig({
+      services: {
+        requestAuthToken: () => Promise.resolve({authToken: 'auth123'}),
+      },
+    }),
+  )
+
+  pollingService.start()
+
+  await sleep(0)
+
+  expect(pollingService.state.context.authToken).toEqual('auth123')
+
+  expect(pollingService.state).toMatchState({pending: 'verifyRetrieval'})
+})
+
+test('it schedules the next poll if lookup failed', async () => {
+  const requestAuthToken = jest.fn().mockRejectedValueOnce(null)
+
+  const pollingService = interpret(
+    authTokenPollingMachine.withConfig({
+      services: {
+        requestAuthToken,
+      },
+    }),
+  )
+
+  pollingService.start()
+
+  await sleep(0)
+
+  expect(pollingService.state).toMatchState({pending: 'scheduleNextPoll'})
+})
+
+test('polls several times for the value', async () => {
```

Comment:
This is the most interesting test. It uses the `SimulatedClock` and `sleep` to both advance the clock and to trigger the Invoked Service. I've left some comments to hopefully make that clear.